autocomplete.go

```diff
@@ -30,51 +30,42 @@ func hasInsensitivePrefix(s string, prefix string) bool {
 	return len(s) >= len(prefix) && strings.EqualFold(s[0:len(prefix)], prefix)
 }
 
-func (a autocomplete) unixAutocomplete(path string) (string, error) {
-	var splittedPath []string
-	if path[0] == '/' {
-		splittedPath = strings.Split(path[1:], "/")
-	} else {
-		splittedPath = strings.Split(path, "/")
+func (a autocomplete) unixAutocomplete(path string) string {
+	if path == "" || path[len(path)-1] == ' '{
+		return path
 	}
-
-	lastValidSplittedPath := splittedPath[:len(splittedPath)-1]
-
-	var lastValidPath string
-	for _, subPath := range lastValidSplittedPath {
-		lastValidPath += "/" + subPath
+	lastSlash := strings.LastIndex(path, "/")
+	if lastSlash == -1 || (path[0] != '/' && path[:2] != "./"){
+		path = "./" + path
+		lastSlash = 1
 	}
-	if lastValidPath == "" {
-		lastValidPath = "/"
-	}
-
-	if !isDir(lastValidPath) {
-		return lastValidPath, nil
+	path = a.findFromPrefix(path, lastSlash)
+	ok, err := isDir(path)
+	if !ok && err == nil {
+		path = path + " "
```

Comment:
I noticed that when it is a file you add a space at the end of the string but, in this case, this program returns the path to a file or folder(in this case a specific file) and if in this returned path the string has an space at the end, I think this may cause some errors in the future when the user will try to manipulate the path. Could you just remove this empty space?

Reply:
Sure!